ov/server_hardware.go

```diff
@@ -689,3 +704,35 @@ func (c *OVClient) SetPowerState(serverHardwareId string, powerState map[string]
 
 	return nil
 }
+
+// Delete the rack server using URI
+func (c *OVClient) DeleteServerHardware(uri utils.Nstring) error {
+	var (
+		hardware ServerHardware
+		err      error
+	)
+
+	hardware, err = c.GetServerHardwareByUri(uri)
+	if err != nil {
+		return err
+	}
+	if hardware.Name != "" {
```

Comment:
why this check? I mean if GetServerHardwareByUri not returning any error then why we are doing this check...because we just need uri to make a delete call

Reply:
Done